### PR TITLE
Fobnail integration

### PR DESCRIPTION
**Pull request by kotylamichal**
## Overview

This PR introduces `fobnail-attester` integration with the required software

## List of issues

N/A

## Evidence

After running `fobnail-attester` server and connecting the token to USB, attestation work well, and device blink on green LED

```
# fobnail-attester 
Creating CoAP server endpoint using UDP.
Registering CoAP resources.
Entering main loop.
[ 3922.605064] usb 2-2: new full-speed USB device number 15 using xhci_hcd
[ 3922.799942] cdc_eem 2-2:1.0 usb0: register 'cdc_eem' at usb-0000:00:10.0-2, CDC EEM Device, da:7f:03:e0:57:12
Received message: metadata
MAC:  0: D:B9:53:D2:50
SMBIOSv2
Manufacturer >PC Engines<
Product Name >apu2<
Serial Number >1373268<
Received message: metadata
MAC:  0: D:B9:53:D2:50
SMBIOSv2
Manufacturer >PC Engines<
Product Name >apu2<
Serial Number >1373268<
Received message: metadata
MAC:  0: D:B9:53:D2:50
SMBIOSv2
Manufacturer >PC Engines<
Product Name >apu2<
Serial Number >1373268<
Received message: quote
```

## Documentation changes

https://github.com/fobnail/docs/pull/32 - PR with description of attestation workflow on apu2

## Notes/known issues

N/A

## Pre-review checklist

- :bangbang: Document changes made in above template
- :bangbang: Document how the changes work (screenshots, demo, CLI tests, etc)
- :bangbang: Changes added to CHANGELOG
- :bangbang: Version bumped
- :bangbang: Documents wrapped at 80 chars
- [x] Pre-review checks are complete

## Reviewer

- [x] Reviewer checks are complete
- :bangbang: Does the change match the task criteria
- :bangbang: Does it work as explained in `Evidence` section

### CODE REVIEW

File: meta-fobnail-distro/recipes-core/systemd/systemd-conf_%.bbappend [hunk 1]
```diff
@@ -0,0 +1,3 @@
+do_install:append(){
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-core/systemd/systemd-conf_%.bbappend:1]: @kotylamichal why we remove those?

File: meta-fobnail-distro/recipes-core/systemd/systemd_%.bbappend [hunk 2]
```diff
@@ -1,3 +1,7 @@
 # we promote the usage of chrony instead of the timesyncd - remove so it does
 # not wast space
 PACKAGECONFIG:remove = "timesyncd"
+
+do_install:append() {
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-core/systemd/systemd_%.bbappend:5]: @kotylamichal why we remove those?

Also, workdir/image is destination dir, which variable is `D` https://docs.yoctoproject.org/ref-manual/variables.html#term-D

File: meta-fobnail-distro/recipes-fobnail/fobnail-attester/files/fobnail_usb.network [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[Match]
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-fobnail/fobnail-attester/files/fobnail_usb.network:1]: @kotylamichal network files are installed in systemd-conf recipe so I think it should go there

File: meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=1ff626c483f6f10b69ee67274daf0775"
+
+SRC_URI = " \
+    git://github.com/fobnail/fobnail-attester;protocol=https;branch=main \
+    file://0001-Makefile-fix-to-build-in-Yocto.patch \
+    file://fobnail_usb.network \
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb:7]: Lets install network file via systemd-conf recipe

File: meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb [hunk 5]
```diff
@@ -0,0 +1,37 @@
+LICENSE = "Unknown"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=e44b3af4925ec58e9f49b9ff143b5493 \
+                    file://COPYING;md5=faed8f005d476edd3f250599a4bb9a75 \
+                    file://ext/tinydtls/LICENSE;md5=ffb073dbb36e7ec5e091047332f302c5"
+
+SRC_URI = "gitsm://github.com/obgm/libcoap.git;protocol=https;branch=release-4.3.0"
+
+PV = "4.3.0+git${SRCPV}"
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb:8]: @kotylamichal after the recipe name will be changed this will not be necessary, the PV will be 4.3.0

File: meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+--disable-tests \
+--disable-documentation \
+--disable-manpages \
+--disable-dtls \
+--disable-shared \
+--enable-fast-install \
+"
+
+do_configure:prepend() {
+    cd ${S}
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb:27]: @kotylamichal why is this needed?

File: meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    touch ar-lib
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb:28]: @kotylamichal also this, you mentioned some patches that were removed?

File: meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb [hunk 8]
```diff
@@ -0,0 +1,37 @@
+LICENSE = "Unknown"
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb:1]: @kotylamichal please lets add here description and homepage to follow some structure of recipes https://docs.yoctoproject.org/dev-manual/common-tasks.html#locating-and-using-a-similar-recipe

Comment (TomaszAIR) [reply meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb:1]: Also, can't we get LICENSE from https://github.com/obgm/libcoap#license-information ?

File: meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb [hunk 9]
```diff
@@ -0,0 +1,28 @@
+LICENSE = "MIT"
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb:1]: @kotylamichal let's also add DESCRIPTION and HOMEPAGE here

File: meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+NETWORKCONFDIR = "/etc/systemd/network"
+
+INSANE_SKIP:${PN} = "ldflags"
+
+do_install() {
+    install -d ${D}${bindir}
+    install -m 755 ${S}/bin/fobnail-attester ${D}${bindir}
+
+    install -d ${D}${NETWORKCONFDIR}
+    install -m 755 ${WORKDIR}/fobnail_usb.network ${D}${NETWORKCONFDIR}
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb:27]: @kotylamichal while installing it please add some comment in the bbappend file

File: meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+PROVIDES:${PN} += "fobnail-attester"
+
+DEPENDS = "openssl tpm2-tss libcoap qcbor"
+
+PV = "1.0+git${SRCPV}"
+SRCREV = "0b7085ff80a3417d079888c88f10578e8120ce88"
+
+S = "${WORKDIR}/git"
+NETWORKCONFDIR = "/etc/systemd/network"
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb:18]: @kotylamichal we should use bitbake.conf file from poky for path variables https://git.yoctoproject.org/poky/tree/meta/conf/bitbake.conf?h=honister so it should be `$sysconfdir/systemd/network`

File: meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb [hunk 12]
```diff
@@ -0,0 +1,28 @@
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=1ff626c483f6f10b69ee67274daf0775"
+
+SRC_URI = " \
+    git://github.com/fobnail/fobnail-attester;protocol=https;branch=main \
+    file://0001-Makefile-fix-to-build-in-Yocto.patch \
+    file://fobnail_usb.network \
+"
+
+PROVIDES:${PN} += "fobnail-attester"
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb:10]: @kotylamichal PN is already in PROVIDES by default https://docs.yoctoproject.org/ref-manual/variables.html#term-PROVIDES

File: meta-fobnail-distro/recipes-fobnail/qcobr/qcbor_git.bb [hunk 13]
```diff
@@ -0,0 +1,36 @@
+LICENSE = "CLOSED"
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-fobnail/qcobr/qcbor_git.bb:1]: @kotylamichal please add description and homepage, license info here https://github.com/laurencelundblade/QCBOR#copyright-and-license

File: meta-fobnail-distro/recipes-fobnail/qcobr/qcbor_git.bb [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+PV = "1.0+git${SRCPV}"
+SRCREV = "44754f738c6534a4304a83d4c6e97b3d3193d887"
+
+S = "${WORKDIR}/git"
+
+FILES:${PN} += " \
+    /usr/local/include/qcbor \
+"
+
+INSANE_SKIP:${PN}-dev += "dev-elf ldflags"
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-fobnail/qcobr/qcbor_git.bb:15]: @kotylamichal why dev-elf is needed here?

File: meta-fobnail-distro/recipes-fobnail/qcobr/qcbor_git.bb [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+
+SRC_URI = "git://github.com/laurencelundblade/QCBOR;protocol=https;branch=master"
+
+PV = "1.0+git${SRCPV}"
+SRCREV = "44754f738c6534a4304a83d4c6e97b3d3193d887"
+
+S = "${WORKDIR}/git"
+
+FILES:${PN} += " \
+    /usr/local/include/qcbor \
```
Comment (TomaszAIR) [inline meta-fobnail-distro/recipes-fobnail/qcobr/qcbor_git.bb:12]: @kotylamichal we install files in `/usr/local/include` or `$includedir`?

File: meta-fobnail-distro/recipes-core/systemd/systemd-conf_%.bbappend (hunk 1 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-core/systemd/systemd-conf_%.bbappend:1]: `systemd` creates configs that override our fobnail token configuration. I will add an appropriate comment to this append

File: meta-fobnail-distro/recipes-core/systemd/systemd_%.bbappend (hunk 2 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-core/systemd/systemd_%.bbappend:5]: like above, `systemd` creates configs that override our fobnail token configuration. I will add an appropriate comment to this append. `D` macro applied

File: meta-fobnail-distro/recipes-core/systemd/systemd-conf_%.bbappend (hunk 1 above)
Comment (TomaszAIR) [reply meta-fobnail-distro/recipes-core/systemd/systemd-conf_%.bbappend:1]: @kotylamichal we only add network config for usb interface, which config overrides this?

Comment (TomaszAIR) [reply meta-fobnail-distro/recipes-core/systemd/systemd-conf_%.bbappend:1]: Also, it is recommended to prefix config files with number - more here https://www.freedesktop.org/software/systemd/man/systemd.network.html#Description

File: meta-fobnail-distro/recipes-fobnail/fobnail-attester/files/fobnail_usb.network (hunk 3 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/fobnail-attester/files/fobnail_usb.network:1]: Ok, but even if it is strictly related to fobnail attester configuration? moved to `systemd-conf_%.bbappend`

File: meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb (hunk 4 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb:7]: ok, like above

File: meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+LIC_FILES_CHKSUM = "file://LICENSE;md5=e44b3af4925ec58e9f49b9ff143b5493 \
+                    file://COPYING;md5=faed8f005d476edd3f250599a4bb9a75 \
+                    file://ext/tinydtls/LICENSE;md5=ffb073dbb36e7ec5e091047332f302c5"
+
+SRC_URI = "gitsm://github.com/obgm/libcoap.git;protocol=https;branch=release-4.3.0"
```
[the remark this answers is not in the document]
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb:6]: done

File: meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb (hunk 5 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb:8]: done

File: meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb (hunk 6 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb:27]: without that `cd`'s task config, compile and install  failed because of path is incorrect - bitbake try to execute that in ${WORKDIR}

Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb:27]: Added comment with description

File: meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb (hunk 7 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb:28]: `autogen.sh` need this empty file - it is described here: https://github.com/obgm/libcoap/commit/0f76881802af5e06e6ee69e07e6c8aca7598adbf 

this bugfix is not available in 4.3.0 so it need to be added in configure prepend. Added comment with description

File: meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb (hunk 8 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/libcoap/libcoap_%.bb:1]: Description and homepage added, license too - it is `BSD 2-Clause “Simplified” License`

File: meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb (hunk 10 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb:27]: done

File: meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb (hunk 11 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb:18]: done, but it was moved to `systemd-conf_%.bbappend`

File: meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb (hunk 12 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/fobnail-attester/fobnail-attester_git.bb:10]: removed

File: meta-fobnail-distro/recipes-fobnail/qcobr/qcbor_git.bb (hunk 13 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/qcobr/qcbor_git.bb:1]: done

File: meta-fobnail-distro/recipes-fobnail/qcobr/qcbor_git.bb (hunk 14 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/qcobr/qcbor_git.bb:15]: It was added to avoid errors with generated `libqcbor.so` file, but now we use `libqcbor.a` so we can add `dev-elf` flag

File: meta-fobnail-distro/recipes-fobnail/qcobr/qcbor_git.bb (hunk 15 above)
Comment (kotylamichal) [reply meta-fobnail-distro/recipes-fobnail/qcobr/qcbor_git.bb:12]: fixed
